### PR TITLE
Skip builds for ignored items

### PR DESCRIPTION
**Pull request by CyanDevs**
This PR enables CI to skip tests for future PRs that _only_ contain changes to dirs/files irrelevant to PR tests. These dirs/files include:
```
   '.jenkins/infrastructure/',
    '.jenkins/docker/',
    '.azure_pipelines/',
    '.github/',
    'doc/',
    '.gitignore',
    'arch.png',
    'BUILDING.md',
    'CONTRIBUTING.md',
    'DOCKER_IMAGES.md',
    'LICENSE',
    'owners.txt',
    'README.md',
    'VERSION'
```
Should dirs/files be added/removed from this list, please suggest it here.

Other minor changes include:
* Fix bug where committer emails weren't found and used for notifications. This will restore build notifications.
* Remove rs-- and Sahakait from committers
* Clean up unused pipeline file: .jenkins/tests.Jenkinsfile

Signed-off-by: Chris Yan <chrisyan@microsoft.com>

### CODE REVIEW

**Review by radhikaj — APPROVED** [review]

File: .jenkins/pipelines/pr.Jenkinsfile [hunk 1]
```diff
@@ -10,12 +10,30 @@ APPROVED_AUTHORS = [
...
     'vtikoo'
 ]
 
+// Do not trigger a full CI build if changes contain only these ignored files
+IGNORED_DIRS = [
+    '.jenkins/infrastructure',
+    '.jenkins/docker',
+    '.azure_pipelines',
+    '.github',
+    'doc'
```
Comment (RRathna) [inline .jenkins/pipelines/pr.Jenkinsfile:23]:
```suggestion
    'docs'
```

File: .jenkins/pipelines/pr.Jenkinsfile [hunk 2]
```diff
@@ -10,12 +10,30 @@ APPROVED_AUTHORS = [
...
+IGNORED_FILES = [
+    '.gitignore',
+    'arch.png',
+    'BUILDING.md',
+    'CONTRIBUTING.md',
+    'DOCKER_IMAGES.md',
+    'LICENSE',
+    'owners.txt',
+    'README.md',
+    'VERSION'
```
Comment (RRathna) [inline .jenkins/pipelines/pr.Jenkinsfile:34]: Also is it possible to add sample READMEs to this list?

File: .jenkins/pipelines/pr.Jenkinsfile (hunk 1 above)
Comment (CyanDevs) [reply .jenkins/pipelines/pr.Jenkinsfile:23]: It is just called `doc` in this repository, so no change is needed here

File: .jenkins/pipelines/pr.Jenkinsfile (hunk 2 above)
Comment (CyanDevs) [reply .jenkins/pipelines/pr.Jenkinsfile:34]: Yes, good suggestion, thanks!

**Pull request merged**